.github/workflows/maven-deploy-to-maven-central.yml

```diff
@@ -5,8 +5,8 @@
 name: Generate and Deploy to Sonatype
 
 on:
-  push:
-    branches: [ main ]
+  release:
```

Comment:
This does not match what is in the description, i.e. that a push on main triggers a release to maven central. However, I agreed that we should only trigger releases upon tags to ensure we have properly archived the code base for each release.